vernac/declare.ml

```diff
@@ -1068,6 +1068,42 @@ module State = struct
 
   let all pm = ProgMap.bindings pm |> List.map (fun (_,v) -> CEphemeron.get v)
   let find m t = ProgMap.find_opt t m |> Option.map CEphemeron.get
+
+  module View = struct
+    module Obl = struct
+      type t = {
+        name : Id.t
+      ; loc : Loc.t option
+      ; status : bool * Evar_kinds.obligation_definition_status
+      ; solved : bool
+      }
+
+      let make (o : Obligation.t) =
+        let { obl_name; obl_location; obl_status; obl_body; _ } = o in
+        { name = obl_name
+        ; loc = fst obl_location
+        ; status = obl_status
+        ; solved = Option.has_some obl_body
```

Comment:
Do we not want to view the contents of solved obligations?

Reply:
For now we have the loc, but indeed that's an interesting point. I'm sure the API here will evolve but I need more feedback from the UI side, already it feels quite new to see both goals and bullets.